app/src/main/java/com/jshvarts/todoapp/notelist/NoteListViewModel.kt

```diff
@@ -1,58 +1,61 @@
 package com.jshvarts.todoapp.notelist
 
-import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.viewModelScope
 import com.jshvarts.todoapp.arch.MviViewModel
+import com.jshvarts.todoapp.arch.UiAction
 import com.jshvarts.todoapp.arch.UiResult
 import com.jshvarts.todoapp.arch.asUiResult
+import com.jshvarts.todoapp.data.Note
 import com.jshvarts.todoapp.data.NoteRepository
-import com.jshvarts.todoapp.notelist.ui.NoteListUiAction
-import com.jshvarts.todoapp.notelist.ui.NoteListUiEffect
-import com.jshvarts.todoapp.notelist.ui.NoteListUiState
+import com.jshvarts.todoapp.notelist.ui.*
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.*
-import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-private const val SAVED_STATE_HANDLE_KEY = "NoteListViewModel_uiState_Key"
-
 @HiltViewModel
 class NoteListViewModel @Inject constructor(
-  private val savedStateHandle: SavedStateHandle,
-  private val noteRepository: NoteRepository
-) : MviViewModel<NoteListUiAction, NoteListUiState>(savedStateHandle),
+  noteRepository: NoteRepository
+) : MviViewModel<UiAction, NoteListUiState>(),
   MviViewModel.EffectProducer<NoteListUiEffect> {
 
-  override val initialState: NoteListUiState
-    get() = NoteListUiState.Loading
+  override val initialState: NoteListUiState = NoteListUiState(
+    pendingTodosUiState = NoteListPendingTodosUiState.Loading,
+    completedTodosUiState = NoteListCompletedTodosUiState.Loading
+  )
 
-  override val savedStateHandleKey: String
-    get() = SAVED_STATE_HANDLE_KEY
+  override val savedStateHandleKey = null
 
-  override fun handleAction(action: NoteListUiAction) {
-    when (action) {
-      NoteListUiAction.LoadList -> onLoadList()
-    }
-  }
+  override fun handleAction(action: UiAction) = Unit
```

Comment:
so for this screen (so far) it appears that actions are not needed at all since all data is loaded on view model initialization.  This syntax makes actions irrelevant since they are missing

Actions will be added in the next PR when I add pull to refresh.